doc/source/dev/roadmap-detailed.rst

```diff
@@ -123,8 +123,26 @@ feature requests.
 
 constants
 `````````
-This module is basically done, low-maintenance and without open issues.
+This module needs only periodic updates to the numerical values.
 
+differentiate
+`````````````
+This module was added with the understanding that its scope would be limited.
+The goal is to provide basic differentiation of black-box functions, not
+replicate all features of existing differentiation tools. With that in mind,
+items for future work include:
+
+- Improve support for functions that accept additional arguments (e.g. add
+  ``*args`` to ``jacobian`` and ``hessian``). Note that this is not trivial
+  due to the way elements of the arrays are eliminated when their corresponding
+  calculations have converged.
+- Improve implementation of `scipy.differentiate.hessian`: rather than chaining
+  first-order differentiation, use a second-order differentiation stencil.
+- Consider the addition of an option to use relative step sizes.
+- Consider generalizing the approach to use "polynomial extrapolation"; i.e.,
```

Comment:
"polynomial extrapolation" was the term used in gh-17059 for this idea, AFAICT. Regardless of what the implementation *looks* like or how the theory is presented (Richardson extrapolation, etc...) , I think finite step numerical differentiation is equivalent to fitting a polynomial to the data and differentiating the polynomial exactly. A key difference between [Jacobi](https://hdembinski.github.io/jacobi/) from what we currently do is that our method is equivalent to exactly fitting a polynomial of order $n$ with $n+1$ points whereas Jacobi uses `polyfit` to fit a polynomial of order $n$ with $n + m$. This is discussed a bit in mdhaber/jacobi#1,